File: source/nestjs-petclinic-rest/src/repository/pet-type.repository.ts
import { Repository, EntityRepository } from 'typeorm';
import { CustomRepository } from './../typeorm/typeorm-ex.decorator';
import { PetType } from './../domain/entities/pet-type.entity';
import { IntentMerge } from './../intent/intent.decorators';

@IntentMerge()
@CustomRepository(PetType)
export class PetTypeRepository extends Repository<PetType> {
}

File: source/spring-petclinic-rest/src/main/java/com/spring_petclinic/spring_petclinic_rest/data/PetTypeRepository.java
package com.spring_petclinic.spring_petclinic_rest.data;

import org.springframework.data.jpa.repository.JpaRepository;
import com.spring_petclinic.spring_petclinic_rest.domain.models.PetType;
import com.spring_petclinic.spring_petclinic_rest.intent.IntentMerge;
import java.util.UUID;

/**
 * Spring Data JPA repository for the PetType entity.
 */
@IntentMerge
public interface PetTypeRepository extends JpaRepository<PetType, Integer> {
}